src/TTextEdit.cpp

```diff
@@ -466,6 +497,13 @@ void TTextEdit::drawLine(QPainter& painter, int lineNumber, int lineOfScreen, in
         drawGraphemeForeground(painter, fgColors.at(index), textRects.at(index), graphemes.at(index), charStyle);
         indexOfChar = nextBoundary;
     }
+
+    // If caret mode is enabled and the line is empty, still draw the caret.
+    if (mudlet::self()->isCaretModeEnabled() && mCaretLine == lineNumber && lineText.size() == 0) {
+        int charWidth = getGraphemeWidth(getGraphemeBaseCharacter("a"));
```

Comment:
Hm, good question. I don’t remember anymore. I think I did that because I thought that perhaps the width of `a` depends on the locale. For example, in Chinese/Japanese most graphemes have width 2, so perhaps they use width 2 for Latin characters as well?

I don’t know if that is actually the case though. I will try to find out.

Reply:
> ...For example, in Chinese/Japanese most graphemes have width 2...

Yeah the ones with a **W**ide width - which in those locales is pretty much all of them - and some codepoints are the wide equivalents of an unspecified corresponding narrow (normal) width ones - yet some other East-Asion codepoints have an "Ambiguous" width!

Reply:
I tested with the zh_TW locale and even then `charWidth` is 1, so I'm removing this call to `getGraphemeWidth()`. Thank you for noticing this unnecessary piece of code.